### PR TITLE
Adds validation to fix email addresses.

### PR DESCRIPTION
**Pull request by wjames111**
## Description

- Tools - Fix Email Address - Add validation for edited email addresses
-  Ticket #[8162](https://jira.cru.org/browse/MPDX-8162)

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [x] I have requested a review from another person on the project

### CODE REVIEW

File: src/components/Tool/FixEmailAddresses/FixEmailAddressPerson/FixEmailAddressPerson.tsx [hunk 1]
```diff
@@ -308,61 +319,36 @@ export const FixEmailAddressPerson: React.FC<FixEmailAddressPersonProps> = ({
...
-                                <>
+                      <Formik
+                        key={index}
+                        enableReinitialize={true}
+                        initialValues={{
+                          newEmail: email.email,
+                        }}
+                        validationSchema={validationSchema}
+                        onSubmit={async (values) => {
+                          await handleChange(id, index, values.newEmail);
```
Comment (canac) [inline src/components/Tool/FixEmailAddresses/FixEmailAddressPerson/FixEmailAddressPerson.tsx:330]: `handleChange` doesn't return a promise.

```suggestion
                        onSubmit={(values) => {
                          handleChange(id, index, values.newEmail);
```

File: src/components/Tool/FixEmailAddresses/FixEmailAddressPerson/FixEmailAddressPerson.tsx [hunk 2]
```diff
@@ -372,88 +358,153 @@ export const FixEmailAddressPerson: React.FC<FixEmailAddressPersonProps> = ({
...
+                                      setFieldValue('newEmail', e.target.value);
+                                      handleSubmit();
+                                    }}
+                                    disabled={email.source !== appName}
+                                  />
+                                  <FormHelperText
+                                    error={true}
+                                    data-testid="statusSelectError"
+                                  >
+                                    {errors.newEmail && errors.newEmail}
```
Comment (canac) [inline src/components/Tool/FixEmailAddresses/FixEmailAddressPerson/FixEmailAddressPerson.tsx:462]:
```suggestion
                                    {errors.newEmail}
```

**Review by canac — APPROVED** [review]
> The only real issue I found with this way is that the confirm button isn’t blocked when an illformed address is entered.

I saw a few times where typing an invalid email address and clicking Confirm sent the invalid email in the GraphQL mutation. But it appears that the server ignored it without returning an error. It's a little weird but probably OK.

It looks like we have a form inside of a form now. I'm not sure if that's bad or not, but if it needs to be changed, we can do that when we research performance.

**Comment by wrandall22** [comment]
> It looks like we have a form inside of a form now. I'm not sure if that's bad or not

Nested forms are not supported in HTML technically.

**Comment by canac** [comment]
> I'm not seeing the nested form, am I missing something?

You're right, I wasn't reading carefully. I thought `FixEmailAddresses` was using Formik, but it's just using a simple `useState` (which I think is the right choice).

I have no issues with the code you have here. Sorry for slowing this down!

**Pull request merged**